src/app/chart/donut/donut-config.ts

```diff
@@ -0,0 +1,36 @@
+import { ChartConfig } from '../chart-config';
+/**
+ * A config containing properties for the sparkline chart
+ */
+export class DonutConfig extends ChartConfig {
+
+  /**
+   * An optional function to customize the text of the center label
+   */
+  centerLabelFn?: any;
+
+  /**
+   * An optional function to handle when donut is clicked
+   */
+  onClickFn?: any;
```

Comment:
Instead of a callback function, we would like to have an onClick event in the component.

Reply:
This onClickFn mirrors `angular-patternfly` and would be an API difference between `angular-patternfly` and `patternfly-ng` if changed, is that okay? My approach was to keep things one to one such that anyone deciding to port from one to the other would not have to make any config changes due to incompatibilities. However I realize `angularjs` and `angular` are completely different projects so this might not be a concern for this project.

I don't understand what an "onClick event in the component" means, could you elaborate?

Reply:
Yes, it's ok to be different from APF. This is our opportunity to clean things up.

Instead of calling a function when the donut is clicked, we emit an event. A developer can easily subscript to the event like so:

<pfng-donut-chart (onClick)="myFunction($event)"...

Then, in the component we would have something like this:

  /**
   * The event emitted when an item has been clicked
   */
  @Output('onClick') onClick = new EventEmitter();
And some method to handle the click:

handleClick($event: MouseEvent): void {
&nbsp;&nbsp;this.onClick.emit({
&nbsp;&nbsp;&nbsp;&nbsp;item: item
&nbsp;&nbsp;} as ListEvent);
}